client/src/pages/ProfileListings/DateSelectPopover.tsx

```diff
@@ -0,0 +1,135 @@
+import { useState } from 'react';
+import Grid from '@material-ui/core/Grid';
+import Snackbar from '@material-ui/core/Snackbar';
+import Card from '@material-ui/core/Card';
+import CardHeader from '@material-ui/core/CardHeader';
+import CardActions from '@material-ui/core/CardActions';
+import Popover from '@material-ui/core/Popover';
+import CloseIcon from '@material-ui/icons/Close';
+import Typography from '@material-ui/core/Typography';
+import IconButton from '@material-ui/core/IconButton';
+import Button from '@material-ui/core/Button';
+import { MuiPickersUtilsProvider, KeyboardDatePicker } from '@material-ui/pickers';
+import DateFnsUtils from '@date-io/date-fns';
+import useStyles from './useStyles';
+import Alert from './alert';
+
+interface DateSelectProps {
+  open: true | false;
+  handleOpen: any;
+  handleUpdate: any;
+}
+
+const DateSelectPopover: React.FC<DateSelectProps> = ({ open, handleOpen, handleUpdate }) => {
+  const classes = useStyles();
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [dateFrom, setDateFrom] = useState<Date | null>(null);
+  const [dateTo, setDateTo] = useState<Date | null>(null);
+
+  const handleDateFromChange = (date: Date | null) => {
+    setDateFrom(date);
+  };
+
+  const handleDateToChange = (date: Date | null) => {
+    if (date && dateFrom && date < dateFrom) {
+      setSnackbarOpen(true);
+    } else {
+      setDateTo(date);
+    }
+  };
+
+  const updateDateRange = (dateFrom: Date | null, dateTo: Date | null) => {
+    dateFrom && dateTo ? handleUpdate(dateFrom, dateTo) : handleUpdate(null);
```

Comment:
```suggestion
    dateFrom && dateTo ? handleUpdate(dateFrom, dateTo) : handleUpdate(null, null);
```
Because of the above change in the type of `handleUpdate`, we need to specify two arguments.